utils/vlog/vlog_to_test_eblif.py

```diff
@@ -0,0 +1,141 @@
+    .inputs A B
+    .outputs C D
+    .subckt mod A=A B=B C=C D=D
+    .end
+    >>> print(blif('mod', [('A', 2)], [('C', 2)]))
+    .model top
+    .inputs A0 A1
+    .outputs C0 C1
+    .subckt mod A[0]=A0 A[1]=A1 C[0]=C0 C[1]=C1
+    .end
+    """
+    inputs = list(flatten(inputs))
+    outputs = list(flatten(outputs))
+
+    return """\
+.model top
+.inputs {inputs}
+.outputs {outputs}
+.subckt {name} {iomap}
+.end""".format(
+            name=name,
+            inputs=" ".join(s for s, d in inputs),
+            outputs=" ".join(s for s, d in outputs),
+            iomap=" ".join("{}={}".format(d, s) for s,d in inputs+outputs),
+        )
+
+
+parser = argparse.ArgumentParser()
+__doc___="""\
+Generate an arch.xml file which includes pb_type.xml and model.xml files for
+testing with Verilog to Routing.
+"""
+
+parser.add_argument(
+    'infiles',
+    metavar='input.v',
+    type=str,
+    nargs='+',
+    help="""\
+One or more Verilog input files, that will be passed to Yosys internally.
+They should be enough to generate a flattened representation of the model,
+so that paths through the model can be determined.
+"""
+)
+parser.add_argument(
+    '--top',
+    help="""\
+Top level module, will usually be automatically determined from the file name
+%.sim.v
+"""
+)
+parser.add_argument(
+    '--includes',
+    help="""\
+Command separate list of include directories.
+""",
+    default=""
+)
+parser.add_argument(
+    '-o',
+help="""\
+Output filename, default 'model.xml'
+""")
+
+
+def main(args):
+    args = parser.parse_args(args)
+
+    iname = os.path.basename(args.infiles[0])
+
+    outfile = "{}.test.eblif".format(iname)
+    if "o" in args and args.o is not None:
+        outfile = args.o
+    outfile = os.path.abspath(outfile)
+
+    if args.includes:
+        for include in args.includes.split(','):
+            yosys.run.add_include(include)
+
+    aig_json = yosys.run.vlog_to_json(args.infiles, flatten=True, aig=True)
+
+    if args.top is not None:
+        yj = YosysJSON(aig_json, args.top)
+        top = yj.top
+    else:
+        wm = re.match(r"([A-Za-z0-9_]+)\.sim\.v", iname)
+        if wm:
+            top = wm.group(1).upper()
+        else:
+            print(
+                "ERROR file name not of format %.sim.v ({}), cannot detect top level. Manually specify the top level module using --top"
+            ).format(iname)
+            return 1
+        yj = YosysJSON(aig_json, top)
+
+    tmod = yj.top_module
+
+    inputs = []
+    outputs = []
+    for name, width, bits, iodir in tmod.ports:
+        if iodir == "input":
+            inputs.append((name, width))
+        elif iodir == "output":
+            outputs.append((name, width))
+        else:
+            assert False, "Unsupported iodir {}".format(iodir)
+
+    eblif = blif(top, inputs, outputs)
+    with open(outfile, "w") as f:
+        f.write(eblif)
+
+    return 0
+
+
+
+
+if __name__ == "__main__":
+    import doctest
+    doctest.testmod()
```

Comment:
I don't think it makes sense to run doctest every time you run the main function?